From c34f2067b8efed229d773e3a0884c9cf02ae0513 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Tue, 13 Oct 2015 15:08:46 +0800
Subject: [PATCH 0559/5242] MLK-11699-1 ARM: dts: imx6sx-sabreauto: add dts
 support for backlight and pwm4

commit  38bd2b84d1560be6f3c37cf40c8ea3027429169a from
https://source.codeaurora.org/external/imx/linux-imx.git

Add dts support for backlight and pwm4 modules on
imx6sx sabreauto board.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sabreauto.dts |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx-sabreauto.dts b/arch/arm/boot/dts/imx6sx-sabreauto.dts
index c4917d1..f78c171 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto.dts
@@ -10,6 +10,14 @@
 	model = "Freescale i.MX6 SoloX Sabre Auto Board";
 	compatible = "fsl,imx6sx-sabreauto", "fsl,imx6sx";
 
+	backlight2 {
+		compatible = "pwm-backlight";
+		pwms = <&pwm4 0 5000000>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <6>;
+		fb-names = "mxs-lcdif1";
+	};
+
 	clocks {
 		codec_osc: anaclk2 {
 			#clock-cells = <0>;
@@ -379,6 +387,12 @@
 	status = "okay";
 };
 
+&pwm4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm4_0>;
+	status = "okay";
+};
+
 &iomuxc {
 	pinctrl_egalax_int: egalax-intgrp {
 		fsl,pins = <
@@ -515,6 +529,12 @@
 		>;
 	};
 
+	pinctrl_pwm4_0: pwm4grp-0 {
+		fsl,pins = <
+			MX6SX_PAD_SD1_DATA1__PWM4_OUT 0x110b0
+		>;
+	};
+
 	pinctrl_spdif_3: spdifgrp-3 {
 		fsl,pins = <
 			MX6SX_PAD_ENET2_COL__SPDIF_IN           0x1b0b0
-- 
1.7.9.5

